### PR TITLE
Use our fork of nanojson

### PR DESCRIPTION
**Pull request by wb9688**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

### CODE REVIEW

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import java.text.SimpleDateFormat;
 import java.util.Calendar;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
```
Comment (TobiGr) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java:25]: unused

Comment (wb9688) [reply extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java:25]: Yup, I used that for getting JSON that nanojson fails to parse without spamming YouTube's servers.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMusicSearchExtractor.java [hunk 2]
```diff
@@ -301,16 +294,7 @@ public String getUploaderUrl() throws ParsingException {
...
-                                    return null;
-                                }
-
-                                final String url = getUrlFromNavigationEndpoint(navigationEndpoint);
-
-                                if (url != null && !url.isEmpty()) {
-                                    return url;
-                                }
-                                throw new ParsingException("Could not get uploader url");
+                                return getUrlFromNavigationEndpoint(navigationEndpoint);
```
Comment (TobiGr) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMusicSearchExtractor.java:297]: `getUrlFromNavigationEndpoint(navigationEndpoint)` returns null when navigationEndpoint is an empty Object or YouTube changed something in their JSON. In this case, we could not get the uploader URL and should throw a ParsingException. So please keep the null check for the URL.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java [hunk 3]
```diff
@@ -98,7 +93,7 @@ public String getUrl() throws ParsingException {
     @Override
     public String getName() throws ParsingException {
         String name = getTextFromObject(videoInfo.getObject("title"));
-        if (name != null && !name.isEmpty()) return name;
+        if (name == null || !name.isEmpty()) return name;
```
[the remark this answers is not in the document]
Comment (wb9688) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java:96]: This change is indeed an error, and `name` could indeed not be `null`. There are probably other usages of `getTextFromObject()` where that check could be removed as well. I'll do that tomorrow.

**Pull request merged**